src/ucp/rndv/proto_rndv.inl

```diff
@@ -80,8 +83,12 @@ ucp_proto_rndv_ats_handler(void *arg, void *data, size_t length, unsigned flags)
         }
     }
 
+    ucp_datatype_iter_mem_dereg(req->send.ep->worker->context,
+                                &req->send.state.dt_iter, UCP_DT_MASK_ALL);
+    ucp_datatype_iter_cleanup(&req->send.state.dt_iter, UCP_DT_MASK_ALL);
```

Comment:
maybe use  ucp_proto_request_zcopy_cleanup() ?
or add dt_mask parameter to ucp_proto_request_zcopy_complete?